crates/iceberg-catalog/src/service/authz/mod.rs

```diff
@@ -131,6 +131,12 @@ impl TableUuid for TableIdentUuid {
     }
 }
 
+impl TableUuid for TabularDetails {
```

Comment:
Should we move the `TableUuid` trait out of authz?
Feels odd to implement it in the authz module for TabularDetails

Reply:
moved TabularDetails, it's also in service/mod.rs now next to TableIdentUuid, would you like the trait to move there as well?